x/dymns/keeper/sell_order.go

```diff
@@ -0,0 +1,140 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	dymnstypes "github.com/dymensionxyz/dymension/v3/x/dymns/types"
+)
+
+// SetSellOrder stores a Sell-Order into the KVStore.
+func (k Keeper) SetSellOrder(ctx sdk.Context, so dymnstypes.SellOrder) error {
+	if err := so.Validate(); err != nil {
+		return err
+	}
+
+	if !so.HasSetSellPrice() {
+		so.SellPrice = nil
+	}
+
+	if so.HighestBid != nil && len(so.HighestBid.Params) == 0 {
+		so.HighestBid.Params = nil
+	}
+
+	// persist record
+	store := ctx.KVStore(k.storeKey)
+	soKey := dymnstypes.SellOrderKey(so.AssetId, so.AssetType)
+	bz := k.cdc.MustMarshal(&so)
+	store.Set(soKey, bz)
+
+	ctx.EventManager().EmitEvent(so.GetSdkEvent(dymnstypes.AttributeValueSoActionNameSet))
+
+	return nil
+}
+
+// GetSellOrder retrieves active Sell-Order of the corresponding Dym-Name/Alias from the KVStore.
+// If the Sell-Order does not exist, nil is returned.
+func (k Keeper) GetSellOrder(ctx sdk.Context,
+	assetId string, assetType dymnstypes.AssetType,
+) *dymnstypes.SellOrder {
+	store := ctx.KVStore(k.storeKey)
+	soKey := dymnstypes.SellOrderKey(assetId, assetType)
+
+	bz := store.Get(soKey)
+	if bz == nil {
+		return nil
+	}
+
+	var so dymnstypes.SellOrder
+	k.cdc.MustUnmarshal(bz, &so)
+
+	return &so
+}
+
+// DeleteSellOrder deletes the Sell-Order from the KVStore.
+func (k Keeper) DeleteSellOrder(ctx sdk.Context, assetId string, assetType dymnstypes.AssetType) {
+	so := k.GetSellOrder(ctx, assetId, assetType)
+	if so == nil {
+		return
+	}
```

Comment:
why need to do this?

Reply:
This can happen, if someone jumped in and call existing method like they know it all.
And that's why I treat every method independently.

Btw: I have seen impossible things happened multiple times. Today it's not does not mean it won't happens tmr.